cparser/Elab.ml

```diff
@@ -1175,23 +1199,21 @@ let rec elab_designator loc env zi desig =
       | Some zi' ->
           elab_designator loc env zi' desig'
       | None ->
-          error loc "%s has no member named %s" (I.name zi) name;
+          error loc "field designator '%s' does not refer to any field in type '%s'" name (I.name zi);
```

Comment:
The original message was better:
* First, `I.name zi` returns a description of the current path (e.g. `a[5]` or `a.b.c`), not the description of a type.  
* Second, "member" is the ISO C standard terminology; "field" is not.